models/agent_model.go

```diff
@@ -631,6 +641,15 @@ func (s Agent) TemplateDelimiters(svc *Service) *DelimiterPair {
 	return &tdp
 }
 
+func (s *Agent) buildWebConfigFile(password string) string {
+	buf, err := bcrypt.GenerateFromPassword([]byte(password), passwordHashCost)
+	if err != nil {
+		log.Fatal(err, "cannot calculate hash for password")
+	}
+
+	return fmt.Sprintf("basic_auth_users:\n    pmm: %s\n", string(buf))
```

Comment:
let's use templates instead of fmt.Sprintf, so it's easier to read.

Reply:
Do you mean those templates?
`tmpl, err := template.New("test").Parse("{{.Count}} items are made of {{.Material}}")
if err != nil { panic(err) }
err = tmpl.Execute(os.Stdout, sweaters)
if err != nil { panic(err) }`

Reply:
Or move that 
> basic_auth_users:\n    pmm: %s\n

to external file and load as a resource?

Reply:
please check how we use it in QAN, supervisord or scrape configs. it makes your template easier to read.
from what I see I can't understand quickly that it has some structure and might think that pmm is a value for `basic_auth_users`
```
basic_auth_users:
    pmm: {{ buf }}
```